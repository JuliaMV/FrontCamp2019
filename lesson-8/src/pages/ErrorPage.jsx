import React from 'react';
import PropTypes from 'prop-types';

import Header from 'components/header/Нeader';
import SortPanel from 'components/sortPanel/SortPanel';
import Footer from 'components/footer/Footer';
import ErrorTitle from '../components/errorTitle/ErrorTitle';


const ErrorPage = ({ title }) => (
  <>
    <Header />
    <div style={{ position: 'relative' }}>
      <SortPanel description="7 films found" isFilter />
      <ErrorTitle title={title} />
    </div>
    <Footer />
  </>
);

ErrorPage.propTypes = {
  title: PropTypes.string.isRequired,
};


export default ErrorPage;
