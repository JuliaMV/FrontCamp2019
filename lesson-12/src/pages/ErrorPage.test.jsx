import React from 'react';
import Enzyme, { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import Adapter from 'enzyme-adapter-react-16';
import { ErrorPage } from './ErrorPage';

Enzyme.configure({ adapter: new Adapter() });

const title = 'ERROR TITLE';

describe('Error page', () => {
  test(`should matches the snapshot with title "${title}"`, () => {
    const props = {
      title,
    };

    const component = shallow(<ErrorPage title={props.title} />);
    expect(toJson(component)).toMatchSnapshot();
  });
});
