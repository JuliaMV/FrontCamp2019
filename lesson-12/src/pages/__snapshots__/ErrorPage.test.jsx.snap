// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Error page should matches the snapshot with title "ERROR TITLE" 1`] = `
<Fragment>
  <Header
    activeFilter="title"
    filmsHandler={[Function]}
    filterHandler={[Function]}
  />
  <div
    style={
      Object {
        "position": "relative",
      }
    }
  >
    <SortPanel
      activeFilter=""
      description=""
      filterHandler={[Function]}
      isFilter={true}
    />
    <ErrorTitle
      title="ERROR TITLE"
    />
  </div>
  <Footer />
</Fragment>
`;
